src/libsync/owncloudpropagator.cpp

```diff
@@ -94,6 +94,16 @@ int OwncloudPropagator::hardMaximumActiveJob()
     return max;
 }
 
+PropagateItemJob::~PropagateItemJob()
+{
+    if (auto p = propagator()) {
```

Comment:
Technically PropagatorJob::propagator could use a static_cast and this null-check wouldn't be necessary since `PropagatorJob::PropagatorJob` takes an `OwncloudPropagator *propagator`. Just a nitpick, looks good anyhow.